### PR TITLE
Propagate UserState to PollingMasterInquireClient

### PR DESCRIPTION
**Pull request by bf8086**
Currently `PollingMasterInquireClient` uses a new `Subject` when querying master. If the call needs authentication using credentials from `Subject`, it will fail the request because the credentials are missing. This change add parameters to APIs to allow passing in the required `Subject`.

Manually tested with external `Subject` from Hadoop client.

### CODE REVIEW

**Comment by bf8086** [comment]
@gpang @ggezer PTAL.

**Comment by gpang** [comment]
This should pass in the `UserState`, instead of `Subject`.

**Comment by bf8086** [comment]
Thanks @gpang! I updated the PR. PTAL.

File: minicluster/src/main/java/alluxio/multi/process/MultiProcessCluster.java [hunk 1]
```diff
@@ -673,7 +674,9 @@ public synchronized MasterInquireClient getMasterInquireClient() {
             addresses.add(
                 InetSocketAddress.createUnresolved(address.getHostname(), address.getRpcPort()));
           }
-          return new PollingMasterInquireClient(addresses, ServerConfiguration.global());
+          UserState userState = UserState.Factory.create(ServerConfiguration.global());
```
Comment (gpang) [inline minicluster/src/main/java/alluxio/multi/process/MultiProcessCluster.java]: should this use `ServerUserState`?

File: tests/src/test/java/alluxio/server/ft/MasterFaultToleranceIntegrationTest.java [hunk 2]
```diff
@@ -260,8 +260,9 @@ public void killStandby() throws Exception {
   public void queryStandby() throws Exception {
     List<InetSocketAddress> addresses = mMultiMasterLocalAlluxioCluster.getMasterAddresses();
     Collections.shuffle(addresses);
-    PollingMasterInquireClient inquireClient =
-        new PollingMasterInquireClient(addresses, ServerConfiguration.global());
+    PollingMasterInquireClient inquireClient = new PollingMasterInquireClient(addresses,
+        ServerConfiguration.global(),
+        FileSystemContext.create(ServerConfiguration.global()).getClientContext().getUserState());
```
Comment (gpang) [inline tests/src/test/java/alluxio/server/ft/MasterFaultToleranceIntegrationTest.java]: is this allowed to use `ServerUserState`?

File: minicluster/src/main/java/alluxio/multi/process/MultiProcessCluster.java (hunk 1 above)
Comment (bf8086) [reply minicluster/src/main/java/alluxio/multi/process/MultiProcessCluster.java]: Updated.

File: tests/src/test/java/alluxio/server/ft/MasterFaultToleranceIntegrationTest.java (hunk 2 above)
Comment (bf8086) [reply tests/src/test/java/alluxio/server/ft/MasterFaultToleranceIntegrationTest.java]: It should ideally use a "client" user state but for this test specifically it does not make any difference.

**Comment by bf8086** [comment]
Thanks @gpang! PTAL.

**Comment by gpang** [comment]
alluxio-bot, merge this please

**Comment by gpang** [comment]
@bf8086 does this need to be cherry-picked to `branch-2.0`?

**Comment by bf8086** [comment]
@gpang yeah it looks like the same code path exists in `branch-2.0` as well

**Comment by gpang** [comment]
alluxio-bot, cherry-pick this to branch-2.0 please

**Comment by gpang** [comment]
@bf8086 the auto-cherry-pick doesn't merge cleanly, so you will have to cherry-pick manually.

**Pull request merged**